extension/storage/filestorage/client.go

```diff
@@ -303,3 +317,33 @@ func (c *fileStorageClient) getDbSize() (totalSizeResult int64, dataSizeResult i
 	dataSize := totalSize - int64(dbStats.FreeAlloc)
 	return totalSize, dataSize, nil
 }
+
+// moveFileWithFallback is the equivalent of os.Rename, except it falls back to
+// a non-atomic Truncate and Copy if the arguments are on different filesystems
+func moveFileWithFallback(src string, dest string) error {
+	var err error
+	if err = os.Rename(src, dest); err == nil {
+		return nil
+	}
+
+	// EXDEV is the error code for linking cross-device, we want to continue if we encounter it
+	// other errors, we simply return as-is
+	if !errors.Is(err, syscall.EXDEV) {
+		return err
+	}
+
+	// if we're trying to rename across devices, try truncate and copy instead
+	data, err := os.ReadFile(src) // assuming the file isn't too big
+	if err != nil {
+		return err
+	}
+
+	err = os.Truncate(dest, 0)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(dest, data, 0600)
```

Comment:
Should this error be checked before we attempt to remove src?

Reply:
Yes, it should, good catch!